external/fv3fit/fv3fit/emulation/thermobasis/xarray.py

```diff
@@ -0,0 +1,78 @@
+import dataclasses
+import logging
+from typing import MutableMapping, Hashable, Union
+import xarray as xr
+import tensorflow as tf
+from fv3fit.emulation.thermobasis.batch import to_tensors, to_dict_no_static_vars
+from fv3fit.emulation.thermobasis.emulator import Emulator, Config
+
+State = MutableMapping[Hashable, xr.DataArray]
+
+
+@dataclasses.dataclass
+class XarrayEmulator:
```

Comment:
Between this and Emulator, is there a Predictor class somewhere? I don't see any classes inheriting from Predictor in this PR, though they appear to follow the same APIs. it's not strictly necessary, but it can help make it clear what is going to be plugged in to a prognostic simulation.

Reply:
XarrayEstimator is close to a predictor. I was planning to merge the prognostic run interfaces in a follow-up PR: https://github.com/VulcanClimateModeling/fv3net/blob/f5b2ecb3ec178e1f8524e90dd54d23cc6b49da17/workflows/prognostic_c48_run/runtime/emulator/adapter.py#L40.

I basically use

Reply:
Can you add `Predictor` as a parent class to this, or would it require significant code changes?

Reply:
It's almost a one-liner, but  `Predictor.predict` requires `xr.Dataset` inputs and outputs. This gives this type error:
```
./check_types.sh
external/fv3fit/fv3fit/emulation/thermobasis/xarray.py:35: error: Argument 1 of "predict" is incompatible with supertype "Predictor"; supertype defines the argument type as "Dataset"
external/fv3fit/fv3fit/emulation/thermobasis/xarray.py:35: error: Return type "MutableMapping[Hashable, DataArray]" of "predict" incompatible with return type "Dataset" in supertype "Predictor"
```

We could probably modify this to take `Mapping[Hashable, xr.DataArray]`, but I'll leave that to another PR.